### DISCUSSION TITLE
Multi-step Form Example using Zustand

### DISCUSSION THREAD

**Discussion by theodorusclarence** [react-hook-form/react-hook-form#6382] [Show and tell]
Doing multi-step form is kinda hard, we need to use state management to temporarily store data of each step. It becomes more complex if we have file uploads in the forms.

So, my friend and I decided to attempt creating one. We also add an example of clean file upload using dropzone and lightbox preview.

This site is built using:
- Next.js
- TypeScript
- React Hook Form for form state management, with Yup for validation
- Zustand for storing each step data
- React Leaflet for the map component, layered with google maps

Check the site at https://clarence.link/rhf-stepform
and the GitHub repo at https://github.com/theodorusclarence/rhf-stepform

If you find it useful, a star is appreciated!

And here is the demo:

https://user-images.githubusercontent.com/55318172/131220135-2e66937e-6bd1-4526-8533-09b6be8885aa.mp4

**Comment by aidrecabrera** [comment]
Wtf, I found the perfect solution. Thank you for this. It's really just what I needed! :)

**Comment by ddikodroid** [comment]
thank you so much!

**Comment by IldarReact** [comment]
Very cool code, question is blue outline when clicked, is that a good practice?
